File: src/igb_transition/postprocess/postprocess_hiseq_lane_dual_linux_os.py
#!/usr/bin/env python3
import sys
import os
import subprocess

def main():
    # Check for OS release file
    os_name = "unknown"
    if os.path.isfile("/etc/os-release"):
        with open("/etc/os-release", "r") as f:
            for line in f:
                if line.startswith("ID="):
                    os_name = line.strip().split("=")[1].strip('"')
                    break
    
    print(f"Detected OS: {os_name}")
    
    script_dir = os.path.dirname(os.path.abspath(__file__))
    
    args = sys.argv[1:]
    
    if os_name == 'ubuntu':
        print("Run Ubuntu script")
        script_name = "postprocess_hiseq_lane_ubuntu_22_04_test_gzip.py"
        script_path = os.path.join(script_dir, script_name)
        print(f"{script_name} {' '.join(args)}")
        
        cmd = [sys.executable, script_path] + args
        subprocess.run(cmd, check=True)
        
    elif os_name == 'centos':
        print("Run Centos script")
        script_name = "postprocess_hiseq_lane_centos7_test_gzip.py"
        script_path = os.path.join(script_dir, script_name)
        print(f"{script_name} {' '.join(args)}")
        
        cmd = [sys.executable, script_path] + args
        subprocess.run(cmd, check=True)
        
    else:
        print("Unknown OS")

if __name__ == "__main__":
    main()

File: src/Tools/postprocess/postprocess_hiseq_lane_dual_linux_os.pl

if [ -f /etc/os-release ]; then
    # freedesktop.org and systemd
    . /etc/os-release
    OS=$NAME
    VER=$VERSION_ID
	OS_NAME=$ID
fi
echo $NAME
echo $VERSION_ID
echo $OS_NAME

if [ "$OS_NAME" == 'ubuntu' ]; then
	echo Run Ubuntu script
	echo postprocess_hiseq_lane_ubuntu-22-04_test_gzip.pl $1 $2 $3 $4 $5
	/home/sequser/Facility/Tools/postprocess_hiseq_lane_ubuntu-22-04_test_gzip.pl $1 $2 $3 $4 $5
elif [ "$OS_NAME" == 'centos' ]; then
	echo Run Centos script
	echo postprocess_hiseq_lane_centos7_test_gzip.pl $1 $2 $3 $4 $5
	/home/sequser/Facility/Tools/postprocess_hiseq_lane_centos7_test_gzip.pl $1 $2 $3 $4 $5
else
	echo Unknow OS
fi

